bluetooth/src/commonMain/kotlin/device/DeviceConnectionManager.kt

```diff
@@ -18,43 +18,65 @@
 package com.splendo.kaluga.bluetooth.device
 
 import co.touchlab.stately.collections.sharedMutableMapOf
-import co.touchlab.stately.concurrency.AtomicInt
 import co.touchlab.stately.concurrency.AtomicReference
 import co.touchlab.stately.concurrency.value
 import com.splendo.kaluga.bluetooth.Characteristic
 import com.splendo.kaluga.bluetooth.Descriptor
 import com.splendo.kaluga.bluetooth.Service
+import com.splendo.kaluga.bluetooth.ServiceWrapper
 import com.splendo.kaluga.bluetooth.UUID
 import com.splendo.kaluga.bluetooth.uuidString
-import com.splendo.kaluga.logging.debug
 import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlin.jvm.JvmName
 
 abstract class BaseDeviceConnectionManager(
-    val connectionSettings: ConnectionSettings = ConnectionSettings(),
     val deviceWrapper: DeviceWrapper,
-    val stateRepo: DeviceStateFlowRepo
-) : CoroutineScope by stateRepo {
+    bufferCapacity: Int = BUFFER_CAPACITY,
+    coroutineScope: CoroutineScope
+) : CoroutineScope by coroutineScope {
 
-    private companion object {
-        const val TAG = "DeviceConnectionManager"
+    internal companion object {
+        const val BUFFER_CAPACITY = 256
     }
 
     interface Builder {
         fun create(
-            connectionSettings: ConnectionSettings,
             deviceWrapper: DeviceWrapper,
-            stateRepo: DeviceStateFlowRepo
+            bufferCapacity: Int = BUFFER_CAPACITY,
+            coroutineScope: CoroutineScope
         ): BaseDeviceConnectionManager
     }
 
+    sealed class Event {
+        data class RssiUpdate(val rrsi: Int) : Event()
+        object Connecting : Event()
+        object CancelledConnecting : Event()
+        object Connected : Event()
+        object Disconnecting : Event()
+        data class Disconnected(val onDisconnect: suspend () -> Unit) : Event()
```

Comment:
Callbacks like this are always a bit suspect. I guess this is technically an event and not a state, but looking below does this cause a potential cascading freeze (the second callback wrapped into this one). It is internal to bluetooth so still ok. 

So fine to keep, just noting it in case you have ideas.